silkworm/sync/sync_pos.hpp

```diff
@@ -43,16 +43,20 @@ class PoSSync : public ChainSync {
     Task<void> download_blocks(); /*[[long_running]]*/
 
     // public interface called by the external PoS client
-    Task<rpc::PayloadStatus> new_payload(const rpc::ExecutionPayload&, std::chrono::milliseconds timeout);
-    Task<rpc::ForkChoiceUpdatedReply> fork_choice_update(const rpc::ForkChoiceState&, const std::optional<rpc::PayloadAttributes>&, std::chrono::milliseconds timeout);
+    Task<rpc::PayloadStatus> new_payload(const rpc::NewPayloadRequest& request, std::chrono::milliseconds timeout);
+    Task<rpc::ForkChoiceUpdatedReply> fork_choice_update(const rpc::ForkChoiceUpdatedRequest& request, std::chrono::milliseconds timeout);
     Task<rpc::ExecutionPayloadAndValue> get_payload(uint64_t payloadId, std::chrono::milliseconds timeout);
     Task<rpc::ExecutionPayloadBodies> get_payload_bodies_by_hash(const std::vector<Hash>& block_hashes, std::chrono::milliseconds timeout);
     Task<rpc::ExecutionPayloadBodies> get_payload_bodies_by_range(BlockNum start, uint64_t count, std::chrono::milliseconds timeout);
 
   private:
     static std::shared_ptr<Block> make_execution_block(const rpc::ExecutionPayload& payload);
+    static tl::expected<void, std::string> validate_blob_hashes(const Block& block,
+                                                                const std::optional<std::vector<Hash>>& expected_blob_versioned_hashes);
     void do_sanity_checks(const BlockHeader& header, TotalDifficulty parent_td);
     std::tuple<bool, Hash> has_valid_ancestor(const Hash& block_hash);
+
+    size_t active_chain_validations_{0};
```

Comment:
Is it meant to be thread-safe? We could change it to `std::atomic` if so. Or implement some sort of lock/semaphore.

Reply:
No, `PoSSync` is not meant to be thread-safe by design because it is always called by one thread: the [unique `boost::asio::io_context`](https://github.com/erigontech/silkworm/blob/master/silkworm/sync/sync.cpp#L44) running the internal RPC daemon dedicated to the Engine API end-point.

Such endpoint has always one client only (i.e. the external CL client), so it seems fair to keep the design as simple as possible: no multi-threading, just one `boost::asio::io_context` doing I/O and dispatching the incoming requests to the execution server.